File: src/api/schemas/media.py
from typing import Literal

from pydantic import BaseModel, Field

MediaType = Literal["image", "video"]
MediaSourceType = Literal["upload", "generated", "external_url"]


class MediaAssetResponse(BaseModel):
    id: int
    content_id: int
    media_type: MediaType
    source_type: MediaSourceType
    file_name: str
    file_path: str
    file_url: str
    mime_type: str | None = None
    sort_order: int = 0
    provider: str | None = None
    generation_params: dict | None = None
    created_at: str | None = None


class MediaUploadResponse(MediaAssetResponse):
    pass


class MediaUploadRequest(BaseModel):
    content_id: int = Field(..., gt=0)
    media_type: MediaType

File: frontend/src/api/media.ts
import { api } from './index'

export interface MediaAsset {
  id: number
  content_id: number
  media_type: 'image' | 'video'
  source_type: 'upload' | 'generated' | 'external_url'
  file_name: string
  file_path: string
  file_url: string
  mime_type?: string
  sort_order: number
  provider?: string
  generation_params?: Record<string, unknown> | null
  created_at?: string
}

export async function getContentMedia(contentId: number) {
  const { data } = await api.get<MediaAsset[]>(`/content/${contentId}/media`)
  return data
}

export async function uploadMedia(contentId: number, mediaType: 'image' | 'video', file: File) {
  const formData = new FormData()
  formData.append('content_id', String(contentId))
  formData.append('media_type', mediaType)
  formData.append('file', file)

  const { data } = await api.post<MediaAsset>('/media/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' }
  })
  return data
}

export async function deleteMedia(mediaId: number) {
  const { data } = await api.delete<{ deleted: boolean }>(`/media/${mediaId}`)
  return data
}
